Client/mods/deathmatch/logic/CClientSound.cpp

```diff
@@ -414,6 +414,29 @@ void CClientSound::GetVelocity(CVector& vecVelocity)
     vecVelocity = m_vecVelocity;
 }
 
+void CClientSound::SetLooped(bool bLoop)
+{
+    if (m_bLoop != bLoop)
+    {
+        m_bLoop = bLoop;
+        m_SimulatedPlayPosition.SetLooped(bLoop);
+
+        if (m_pAudio)
+        {
+            double savedPosition = m_pAudio->GetPlayPosition();
+            Destroy();
```

Comment:
Is it really necessary to re-create the audio?

Reply:
After setting "looped" to "false", when the sound is played again, it stops. That's why I have to re-create it because it's mta bug.

Reply:
What if you only recreated it if the sound was finished?

Reply:
Look at the screen. If the sound is played again and the looping state changes, the sound will stop before it finishes, I do not know why. This has been shown on the screen in debugscript.